src/naive/datetime/tests.rs

```diff
@@ -240,3 +240,22 @@ fn test_nanosecond_range() {
         NaiveDateTime::from_timestamp(nanos / A_BILLION, (nanos % A_BILLION) as u32)
     );
 }
+
+#[test]
+#[cfg(feature = "clock")]
+fn test_years_elapsed() {
+    use crate::Utc;
+    const WEEKS_PER_YEAR: f32 = 52.1775;
+
+    // This is always at least one year because 1 year = 52.1775 weeks.
+    let one_year_ago = Utc::today() - Duration::weeks((WEEKS_PER_YEAR * 1.5).ceil() as i64);
+    // A bit more than 2 years.
+    let two_year_ago = Utc::today() - Duration::weeks((WEEKS_PER_YEAR * 2.5).ceil() as i64);
```

Comment:
You should probably not use `today` here but give a specific date. This test is tricky to get it right. @yozhgoor did it right but it isn't easy to understand why.

If you give a specific date, it will be right all the time and it's easier to understand.

Reply:
I mean: the test success depends on the exact date you are running the test. If you get it wrong you can easily get a flaky test that breaks sometimes every 4 years or so.

Reply:
The tests I had here (which I had moved around due to some confusion) are direct ports of the original tests to the new API), so I don't think this is different from before?

Reply:
I mean the test is right and will succeed but it depends on an external variable that is the current date and it is hard to read and understand why it will always succeed.

It's safer and easier to use a specific date (take the current date for instance) so it doesn't vary each time the test is ran.

So my point is not about the test being wrong or right but the maintainability.